src/ORM/Table.php

```diff
@@ -1675,13 +1675,13 @@ protected function _transactionCommitted($atomic, $primary)
      */
     public function findOrCreate($search, callable $callback = null, $options = [])
     {
-        $options += [
+        $options = new ArrayObject($options + [
```

Comment:
Creating arrayobject here and converting it back to array when calling _processFindOrCreate() seem redundant. Wouldn't it be better to create arrayobject later just before dispatching the `afterSaveCommit` event?